client.js

```diff
@@ -1,54 +1,141 @@
 /**
  * @description
- *    This SocketClient uses native WebSocket. It starts a connection and handles messages with the
- *    socket server.
+ *    This SocketClient class uses the native WebSocket object to create a connection to a socket
+ *    server. It creates a connection, handles messages received from the socket server, and can
+ *    send messages to the socket server it is connected to.
+ *
+ *    There are two methods that allow this class to interact with the socket server it is connected
+ *    to. Those methods are:
+ *
+ *        - on(); and
+ *        - to()
+ *
+ *    Use these methods to interact with the connected socket server.
+ *
+ *    There are methods in this class that are prefixed with an underscore. A method prefixed with
+ *    an underscore means it is not meant to be used publicly. Although these methods are not
+ *    private, these methods are mainly for the SocketClient class' use.
+ *
+ *    Helpful links:
+ *    - Learn more about the WebSocket API at the following address:
+ *          https://developer.mozilla.org/en-US/docs/Web/API/WebSocket
+ *    - Learn more about writing web socket client applications at the following address:
+ *          https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_client_applications
  */
 class SocketClient {
+  /**
+   * @description
+   *     Construct an object of this class.
+   *
+   * @return {SocketClient}
+   */
   constructor(options) {
     this.configs = {
       hostname: options.hostname || "localhost",
       port: options.port || "3000",
     };
-    this.listening = {};
+    this.decoder = new TextDecoder();
+    this.listening_to = {};
     this.message_queue = [];
     this.ready = true;
+
+    this._connectToSocketServer();
+    this._listenToSocketServerMessages();
+
+    return this;
+  }
+
+  // FILE MARKER - METHODS FOR PUBLIC USE //////////////////////////////////////////////////////////
+
+  /**
+   * @description
+   *     On receipt of a message in the following channel or on the following event, perform the
+   *     specified callback.
+   *
+   * @param {String} channelOrEvent
+   *     The name of the channel or event.
+   * @param {Function} callback
+   *     The callback to execute on receipt of a message from the channel or event.
+   */
+  on(channelOrEvent, callback) {
+    if (this.connection.readyState === 1) {
+      if (!this.listening_to[channelOrEvent]) {
+        this.listening_to[channelOrEvent] = null;
+      }
+      this.listening_to[channelOrEvent] = callback;
+      const message = JSON.stringify({ listening_to: channelOrEvent });
```

Comment:
actually maybe `listening_to` makes sense... you want to listen to a channel and ask to _be_ a receiver lol